objects/src/notes/file.rs

```diff
@@ -0,0 +1,50 @@
+use vm_core::utils::{ByteReader, ByteWriter, Deserializable, Serializable};
+use vm_processor::DeserializationError;
+
+use super::{Note, NoteDetails, NoteInclusionProof};
+
+// NOTE FILE
+// ================================================================================================
+
+/// A serialized representation of a note.
+pub enum NoteFile {
+    /// The note has not yet been recorded on chain.
+    Details(NoteDetails),
+    /// The note has been recorded on chain.
+    Recorded(Note, NoteInclusionProof),
+}
```

Comment:
I don't love the variant names here. Maybe these could be:

- `DetailsOnly`
- `NoteWithProof`

Reply:
Done.